controllers/scopeinstance_controller.go

```diff
@@ -51,70 +62,362 @@ type ScopeInstanceReconciler struct {
+							"operators.coreos.io/scopeInstance": string(in.GetUID()),
+						},
+						OwnerReferences: []metav1.OwnerReference{{
+							APIVersion: in.APIVersion,
+							Kind:       in.Kind,
+							Name:       in.GetObjectMeta().GetName(),
+							UID:        in.GetObjectMeta().GetUID(),
+						}},
 					},
 					Subjects: cr.Subjects,
 					RoleRef: rbacv1.RoleRef{
-						Kind: "ClusterRole",
-						Name: cr.GenerateName,
+						Kind:     "ClusterRole",
+						Name:     cr.GenerateName,
+						APIGroup: "rbac.authorization.k8s.io",
 					},
 				})
+
 			}
 		}
+	}
 
+	if objects != nil {
+		if err := r.createRBAC(ctx, objects); err != nil {
+			log.Log.Info("Error in creating Role Bindings", "error", err)
+			return err
+		}
 	}
 
+	return nil
+}
+
+func (r *ScopeInstanceReconciler) createRBAC(ctx context.Context, objects []client.Object) error {
+	// Create Objects
 	for _, object := range objects {
-		log.Log.Info("Creating object", "object", object)
+
+		log.Log.Info("Creating object")
 		if err := r.Client.Create(ctx, object); err != nil {
+			if k8sapierrors.IsAlreadyExists(err) {
+				log.Log.Info("RBAC is already Present", "Object", object.GetName())
+				continue
+			}
+
 			log.Log.Error(err, "Error creating object", "object", object)
+			return err
 			/// TODO: try to create all the (cluster)roleBindings before returning an aggregate error.
 		}
 	}
+	return nil
+}
 
-	log.Log.Info("No ScopeInstance error")
+func (r *ScopeInstanceReconciler) updateRBAC(ctx context.Context, objects []client.Object) error {
+	// Update Objects
+	for _, object := range objects {
 
-	return ctrl.Result{}, nil
+		log.Log.Info("Updating object")
+		if err := r.Client.Update(ctx, object); err != nil {
+			log.Log.Error(err, "Error Updating object", "object", object)
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ScopeInstanceReconciler) ensureUpdateOfRBAC(ctx context.Context, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate, clusterRoleBindingMap map[string]rbacv1.ClusterRoleBinding, roleBindingMap map[string]rbacv1.RoleBinding) error {
+
+	listUpdateObjects := []client.Object{}
+	// it will create clusterrole as shown below if no namespace is provided
+	for _, cr := range st.Spec.ClusterRoles {
+
+		if obj, found := clusterRoleBindingMap[cr.GenerateName]; found {
+
+			updateObjects := &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: cr.GenerateName + "-",
+					Labels: map[string]string{
+						"operators.coreos.io/scopeInstance": string(in.GetUID()),
+					},
+				},
+				Subjects: cr.Subjects,
+				RoleRef: rbacv1.RoleRef{
+					Kind:     "ClusterRole",
+					Name:     cr.GenerateName,
+					APIGroup: "rbac.authorization.k8s.io",
+				},
+			}
+
+			if util.IsOwnedByLabel(obj.DeepCopy(), in) &&
+				obj.RoleRef == updateObjects.RoleRef &&
+				reflect.DeepEqual(updateObjects.Subjects, obj.Subjects) {
+				r.logger.Info("Existing ClusterRoleBinding does not need to be updated")
+				return nil
+			}
```

Comment:
This doesn't check that the OwnerReferences and RoleRefs haven't been updated.

Reply:
Already added in `util/util.go`